nab/labeler.py

```diff
@@ -194,52 +208,62 @@ def write(self, destPath):
 
 
   def combine(self):
-    """Combine the raw user labels, and set the anomaly windows."""
-    self.getUserLabels()
-    self.combineRawLabels()
+    """Combine raw and known labels in anomaly windows."""
+    self.getRawLabels()
+    self.combineLabels()
     self.applyWindows()
     self.checkWindows()
-    self.addKnownLabels()
 
 
-  def getUserLabels(self):
-    """Collect the raw user labels from default labels directory."""
+  def getRawLabels(self):
+    """Collect the raw user labels from specified directory."""
     labelPaths = absoluteFilePaths(self.labelDir)
+    self.userLabels = []
+    self.knownLabels = []
+    for path in labelPaths:
+      if "known" in path:
+        self.knownLabels.append(CorpusLabel(path, self.corpus))
+      else:
+        self.userLabels.append(CorpusLabel(path, self.corpus))
 
-    self.userLabels = [CorpusLabel(path, self.corpus) for path in labelPaths]
-
-    if len(self.userLabels) == 0:
+    self.nLabelers = len(self.userLabels)
+    if self.nLabelers == 0:
       raise ValueError("No users labels found")
 
-    self.nlabelers = len(self.userLabels)
-
     
-  def combineRawLabels(self):
+  def combineLabels(self):
```

Comment:
Agreed, I think I'll make separate functions for "bucketing" and "merging" timestamps.